core/runtime/src/main/java/io/quarkiverse/operatorsdk/runtime/QuarkusKubernetesDependentResourceConfig.java

```diff
@@ -1,16 +1,18 @@
 package io.quarkiverse.operatorsdk.runtime;
 
+import java.util.Set;
+
 import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependentResourceConfig;
 import io.quarkus.runtime.annotations.RecordableConstructor;
 
 public class QuarkusKubernetesDependentResourceConfig extends KubernetesDependentResourceConfig {
 
     @RecordableConstructor
-    public QuarkusKubernetesDependentResourceConfig(String[] namespaces, String labelSelector) {
+    public QuarkusKubernetesDependentResourceConfig(Set<String> namespaces, String labelSelector) {
         super(namespaces, labelSelector);
     }
 
-    public String[] getNamespaces() {
+    public Set<String> getNamespaces() {
```

Comment:
I've just found that it was not working for me because I deleted these two getters. Can you add a comment saying that the getters are necessary for the recordable constructor?

Reply:
Yes, I have similar comments on other classes, I wish Quarkus would make this explicit… maybe with an annotation.